### PR TITLE
T4762: Add check for show nat if nat config does not exist

### PR DESCRIPTION
**Pull request by sever-sever**
## Change Summary

Op-mode
Add check for `show nat xxx` if nat configuration does not exist
Change XML to use the script `nat.py` for `show nat destination statistics`
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://phabricator.vyos.net/T4762
* https://phabricator.vyos.net/T4763

## Component(s) name

nat
## How to test
Try to show nat without NAT configuration:
```
vyos@r14:~$ show nat source rules 
NAT is not configured
vyos@r14:~$ 
vyos@r14:~$ 
vyos@r14:~$ /usr/libexec/vyos/op_mode/nat.py show_rules --family inet --direction destination --raw
NAT is not configured
vyos@r14:~$ 
```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: src/op_mode/nat.py [hunk 1]
```diff
@@ -277,7 +283,21 @@ def _get_formatted_translation(dict_data, nat_direction, family):
...
+def _verify():
+    config = ConfigTreeQuery()
+    if not config.exists(base):
+        return False
+    return True
+
+
 def show_rules(raw: bool, direction: str, family: str):
+    # check if config exist
+    if not _verify():
```
Comment (dmbaturin) [inline src/op_mode/nat.py]: I believe this should be a decorator rather than duplicate code.

**Review by dmbaturin — CHANGES_REQUESTED** [review]

File: src/op_mode/nat.py (hunk 1 above)
Comment (sever-sever) [reply src/op_mode/nat.py]: @dmbaturin Could you send an example of the code?

Comment (sever-sever) [reply src/op_mode/nat.py]: > I believe this should be a decorator rather than duplicate code.

Done!

**Comment by jestabro** [comment]
Looks good to me.

**Review by jestabro — APPROVED** [review]

**Review by dmbaturin — APPROVED** [review]

**Pull request merged**